velox/expression/tests/CastExprTest.cpp

```diff
@@ -173,6 +173,159 @@ class CastExprTest : public functions::test::CastBaseTest {
     assertEqualVectors(expected, result);
   }
 
+  template <typename T>
+  void testDecimalToFloatCasts(TypePtr type) {
+    // short to short, scale up.
+    auto shortFlat = makeNullableFlatVector<int64_t>(
+        {DecimalUtil::kShortDecimalMin,
+         DecimalUtil::kShortDecimalMin,
+         -3,
+         0,
+         55,
+         DecimalUtil::kShortDecimalMax,
+         DecimalUtil::kShortDecimalMax,
+         std::nullopt},
+        DECIMAL(18, 18));
+    testComplexCast(
+        "c0",
+        shortFlat,
+        makeNullableFlatVector<T>(
+            {-1,
+             // the same DecimalUtil::kShortDecimalMin conversion, checking
+             // floating point diff works on decimals
+             -0.999999999999999999,
+             -0.000000000000000003,
+             0,
+             0.000000000000000055,
+             // the same DecimalUtil::kShortDecimalMax conversion, checking
+             // floating point diff works on decimals
+             0.999999999999999999,
+             1,
+             std::nullopt}));
+
+    auto longFlat = makeNullableFlatVector<int128_t>(
+        {DecimalUtil::kLongDecimalMin,
+         0,
+         DecimalUtil::kLongDecimalMax,
+         HugeInt::build(0xffff, 0xffffffffffffffff),
+         std::nullopt},
+        DECIMAL(38, 5));
+    testComplexCast(
+        "c0",
+        longFlat,
+        makeNullableFlatVector<T>(
+            {-1e33, 0, 1e33, 1.2089258196146293E19, std::nullopt}));
+  }
+
+  template <typename T>
+  void testDecimalToIntegralCastsOutOfBounds(TypePtr type) {
+    VELOX_CHECK(!(std::is_same<int64_t, T>::value));
+    const auto tooSmall =
+        static_cast<int64_t>(std::numeric_limits<int32_t>::min()) - 1;
+    const auto tooBig =
+        static_cast<int64_t>(std::numeric_limits<int32_t>::max()) + 1;
+
+    VELOX_ASSERT_THROW(
+        testComplexCast(
+            "c0",
+            makeFlatVector<int64_t>({0, tooSmall}, DECIMAL(10, 0)),
+            makeFlatVector<T>(0, 0)),
+        fmt::format(
+            "Failed to cast from DECIMAL(10,0) to {}: -2147483649. Out of bounds.",
+            type->name()));
```

Comment:
You can get the Type name from the physical type using `TypeTraits<T>::name`. You don't need the Type argument.

Reply:
These are native types, like int32_t,  hence that does not work.

Reply:
To clarify, you are passing an argument type only for the name right?

Reply:
Changed into TypeKind and made it work